src/odb/src/defin/definReader.cpp

```diff
@@ -72,6 +72,107 @@
+{
+  // -1 is no bits were provided in the scandef
+  if (bits != -1) {
+    return bits;
+  }
+  // -1 means that the bits are not set in the scandef
+  // We need to check if the inst is sequential to decide what is a
+  // reasonable default value
+  dbMaster* master = inst->getMaster();
+  if (master->isSequential()) {
+    // the default number of bits for sequential elements is 1
+    return 1;
+  }
+  // the default number of bits for combinational logic is 0
+  return 0;
+}
+
+dbITerm* findScanITerm(definReader* reader,
+                       dbInst* inst,
+                       const char* pin_name,
+                       const char* common_pin)
+{
+  if (!pin_name) {
+    if (!common_pin) {
+      reader->_logger->error(utl::ODB,
+                             259,
+                             "SCANDEF is missing either component pin or a "
+                             "COMMONSCANPINS for instance {}",
+                             inst->getName());
+      return nullptr;
+    }
+    // using the common pin name
+    return inst->findITerm(common_pin);
+  }
+  return inst->findITerm(pin_name);
+}
+
+std::variant<dbBTerm*, dbITerm*> findScanTerm(definReader* reader,
+                                              dbBlock* block,
+                                              const char* pin_name)
+{
+  dbBTerm* bterm = block->findBTerm(pin_name);
+  if (bterm) {
+    return bterm;
+  }
+
+  dbITerm* iterm = block->findITerm(pin_name);
+  if (iterm) {
+    return iterm;
+  }
+  reader->_logger->error(
+      utl::ODB, 262, "SCANDEF START/STOP pin {} does not exist", pin_name);
+}
+
+void populateScanInst(definReader* reader,
+                      dbBlock* block,
+                      defiScanchain* scan_chain,
+                      dbScanList* db_scan_list,
+                      const char* inst_name,
+                      const char* in_pin_name,
+                      const char* out_pin_name,
+                      int bits)
+{
+  dbInst* inst = block->findInst(inst_name);
+  if (!inst) {
+    reader->_logger->error(
+        utl::ODB, 255, "SCANDEF Inst {} does not exist", inst_name);
+  }
+
+  dbScanInst* scan_inst = db_scan_list->add(inst);
+
+  dbITerm* scan_in
+      = findScanITerm(reader, inst, in_pin_name, scan_chain->commonInPin());
+  if (!scan_in) {
+    reader->_logger->error(utl::ODB,
+                           256,
+                           "SCANDEF IN pin {} does not exist in cell {}",
+                           in_pin_name,
+                           inst_name);
+  }
+
+  dbITerm* scan_out
+      = findScanITerm(reader, inst, out_pin_name, scan_chain->commonOutPin());
+  if (!scan_out) {
+    reader->_logger->error(utl::ODB,
+                           257,
+                           "SCANDEF OUT pin {} does not exist in cell {}",
+                           in_pin_name,
+                           inst_name);
+  }
+
+  scan_inst->setAccessPins({.scan_in = scan_in, .scan_out = scan_out});
```

Comment:
Nice - I tend to forget you can do this now.

Reply:
:)